archinfo/arch.py

```diff
@@ -532,7 +538,7 @@ def register_arch(regexes, bits, endness, my_arch):
             raise ValueError('Invalid Regular Expression %s' % rx)
     #if not isinstance(my_arch,Arch):
     #    raise TypeError("Arch must be a subclass of archinfo.Arch")
-    if not isinstance(bits, int):
+    if not isinstance(bits, integer_types):
```

Comment:
mhm. I hope I did not introduce unwanted behavior here.

Reply:
Shouldn't be. I'd call it a bugfix if anything.